index.js

```diff
@@ -0,0 +1,42 @@
+const inputFile = document.querySelector("#inputFile");
+const outputJson = document.querySelector("#outputJson");
+
+inputFile.addEventListener('change', convertCsvToJson);
+
+function convertCsvToJson(event) {
+  const file = event.target.files[0];
+
+  const reader = new FileReader();
+  reader.onload = function(e) {
+    const content = e.target.result;
+
+    const lines = content.split('\n');
+    const header = lines[0].split(',');
+    const data = [];
+
+    for (let i = 1; i < lines.length - 1; i++) {
```

Comment:
Minimize esses `for` utilizando um `map` combinado com `reduce` que ambos são métodos nativos do JS para manipular arrays.

O método `map` será usado para iterar sobre cada linha do arquivo CSV e o `reduce` será usado para construir um objeto a partir das colunas de cada linha.